src/NLog/Internal/StringHelpers.cs

```diff
@@ -61,5 +59,28 @@ internal static bool IsNullOrWhiteSpace(string value)
             return string.IsNullOrWhiteSpace(value);
 #endif
         }
+
+        internal static string[] SplitAndTrimTokens(this string value, char delimiter)
+        {
+            if (IsNullOrWhiteSpace(value))
+                return ArrayHelper.Empty<string>();
+
+            if (value.IndexOf(delimiter) == -1)
+            {
+                return new[] { value.Trim() };
+            }
+
+            var result = value.Split(new char[] { delimiter }, StringSplitOptions.RemoveEmptyEntries);
+            for (int i = 0; i < result.Length; ++i)
+            {
+                if (char.IsWhiteSpace(result[i][0]) || char.IsWhiteSpace(result[i][result[i].Length - 1]))
```

Comment:
this a "smart" trim, isn't?

Reply:
Nope just my premature optimization (Always expecting the worst from Microsoft). But actually string.Trim() looks good so no need for this.